src/risk_distributions/risk_distributions.py

```diff
@@ -492,22 +492,40 @@ def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, n
             p = p.values
         return p
 
-    def ppf(self, q: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
+    def ppf(self, q: Union[pd.Series, np.ndarray, float, int],
+            q_dist: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
+        """Quantile function using 2 propensities.
+
+        Parameters
+        ---------
+        q :
+            value propensity
+        q_dist :
+            propensity for picking the distribution
+
+        Returns
+        --------
+            Sample from the ensembled distribution.
+        """
         single_val = isinstance(q, (float, int))
         values_only = isinstance(q, np.ndarray)
 
         q, weights = format_call_data(q, self.weights)
+        q_dist, _ = format_call_data(q_dist, self.weights)
 
         computable = weights[(weights.sum(axis=1) != 0) & ~np.isnan(q)].index
-
         x = pd.Series(np.nan, index=q.index)
 
         if not computable.empty:
+            p_bins = np.cumsum(weights.loc[computable], axis=1)
+            choice_index = (q_dist.values[np.newaxis].T > p_bins).sum(axis=1)
```

Comment:
qdist needs to filter to computable (test if call with series containing nans)

Reply:
Done.
Tested with:
`    test_q = np.linspace(0.001, 0.999, num=SIZE)`
`    test_q[500:600] = np.nan`
Code ran successfully, results values are similar but 100 fewer